src/lib.rs

```diff
@@ -244,40 +244,69 @@ mod ios {
 ))]
 mod unix {
     use std::{
-        ffi::OsStr,
-        io::{Error, Result},
+        env,
+        ffi::{OsStr, OsString},
+        io,
+        path::{Path, PathBuf},
         process::{Command, ExitStatus, Stdio},
     };
 
-    use which::which;
+    pub fn that<T: AsRef<OsStr> + Sized>(path: T) -> io::Result<ExitStatus> {
+        let path = path.as_ref();
+        let open_handlers = [
+            ("xdg-open", &[path] as &[_]),
```

Comment:
Since arrays are fixed sized a slice is needed in order to initialize an array containing differently sized arrays. A vector could have also been used.